.github/workflows/ci.yml

```diff
@@ -31,44 +40,17 @@ jobs:
         run: source ~/.profile && lake exe CedarUnitTests
 
   build_and_test_drt:
-    name: Build and test DRT
+    needs: get-branch-name
+    uses: ./.github/workflows/build_and_test_drt_reusable.yml
+    with:
+      cedar_spec_ref: ${{ github.ref }}
+      cedar_policy_ref: ${{ needs.get-branch-name.outputs.branch_name }}
+  
+  test_docker_build:
+    name: Test Docker build
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        toolchain:
-          - stable
     steps:
       - name: Checkout cedar-spec
         uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - name: Install Lean
-        shell: bash
-        run: |
-              wget https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh
-              bash elan-init.sh -y
-      - name: Prepare Rust build
-        run: rustup update ${{ matrix.toolchain }} && rustup default ${{ matrix.toolchain }}
-      - name: cargo fmt (cedar-policy-generators)
-        working-directory: ./cedar-policy-generators
-        run: cargo fmt --all --check
-      - name: cargo fmt (cedar-drt)
-        working-directory: ./cedar-drt
-        run: cargo fmt --all --check
-      - name: cargo fmt (cedar-drt/fuzz/)
-        working-directory: ./cedar-drt/fuzz
-        run: cargo fmt --all --check
-      - name: cargo rustc (cedar-policy-generators)
-        working-directory: ./cedar-policy-generators
-        run: RUSTFLAGS="-D warnings -F unsafe-code" cargo build --verbose
-      - name: cargo test (cedar-policy-generators)
-        working-directory: ./cedar-policy-generators
-        run: cargo test --verbose
-      - name: Build Lean libraries, and build & test cedar-drt/
-        shell: bash
-        run: source ~/.profile && ./build.sh
-
-  run-integration-tests:
-    uses: ./.github/workflows/run_integration_tests_reusable.yml
-    with:
-      cedar_spec_ref: ${{ github.ref }}
\ No newline at end of file
+      - name: Build Docker image
+        run: docker build .
```

Comment:
Not sure if we should consider using the Docker reusable action, like I did in `cedar-examples`.  (I wasn't aware you can just run `docker build` haha; now I'm not sure which is better.)

Reply:
The reusable action makes more sense, especially if we want to publish the image somewhere later.